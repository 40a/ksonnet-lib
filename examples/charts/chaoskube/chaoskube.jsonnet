local chart = import "../chart.libsonnet";
local maintainer = chart.maintainer;

local core = import "../../../kube/core.libsonnet";
local kubeUtil = import "../../../kube/util.libsonnet";
local v1 = core.v1;
local v1beta1 = core.extensions.v1beta1;

local values = import "values.libsonnet";

{
  // Configuration.

  local chartSpec =
    chart.Default("chaoskube", "0.5.0") +
    chart.Description(
      "Chaoskube periodically kills random pods in your Kubernetes cluster.") +
    chart.Home("https://github.com/linki/chaoskube") +
    chart.Source("https://github.com/linki/chaoskube") +
    chart.Maintainer(
      maintainer.Default("Martin Linkhorst", "linki+kubernetes.io@posteo.de")),

  "chart.json": chartSpec,
  "deployment.json":
    local deployment = v1beta1.deployment + kubeUtil.app.v1beta1.deployment;
    local container = v1.container;
    local appContainer =
      container.Default(values.name, containerImage) +
      container.Args([
        "--in-cluster",
        "--interval=%s" % values.interval,
        "--labels=%s" % values.labels,
        "--annotations=%s" % values.annotations,
        "--namespaces=%s" % values.namespaces,
      ] + if !values.dryRun then ["--no-dry-run"] else []) +
      container.Resources(
        container.resources.Requests(
          values.resources.cpu, values.resources.memory) +
        container.resources.Limits(
          values.resources.cpu, values.resources.memory));
    local labels = chart.DefaultLabels(fullname, chartSpec, release);
    deployment.FromContainer(
      fullname, values.replicas, appContainer, labels=labels, podLabels=labels),

  // Data

  local containerImage = chart.ContainerImage(values.image, values.imageTag),
  local release = {
    name: std.extVar("release.name"),
    service: std.extVar("release.service"),
  },
  local fullname = chart.Fullname(release.name, release.service),
}