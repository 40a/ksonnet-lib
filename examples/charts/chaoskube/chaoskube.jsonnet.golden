{
   "chart.json": {
      "apiVersion": "v1",
      "description": "Chaoskube periodically kills random pods in your Kubernetes cluster.",
      "engine": "gotpl",
      "home": "https://github.com/linki/chaoskube",
      "maintainers": [
         {
            "email": "linki+kubernetes.io@posteo.de",
            "name": "Martin Linkhorst"
         }
      ],
      "name": "chaoskube",
      "sources": [
         "https://github.com/linki/chaoskube"
      ],
      "version": "0.5.0"
   },
   "deployment.json": {
      "apiVersion": "extensions/v1beta1",
      "kind": "Deployment",
      "metadata": {
         "annotations": { },
         "labels": {
            "app": "3-tiller",
            "chart": "chaoskube",
            "heritage": "tiller",
            "release": "3"
         },
         "name": "3-tiller"
      },
      "spec": {
         "replicas": 1,
         "template": {
            "metadata": {
               "annotations": { },
               "labels": {
                  "app": "3-tiller",
                  "chart": "chaoskube",
                  "heritage": "tiller",
                  "release": "3"
               }
            },
            "spec": {
               "containers": [
                  {
                     "args": [
                        "--in-cluster",
                        "--interval=10m",
                        "--labels=",
                        "--annotations=",
                        "--namespaces="
                     ],
                     "env": [ ],
                     "image": "quay.io/linki/chaoskube:v0.5.0",
                     "imagePullPolicy": "Always",
                     "name": "chaoskube",
                     "ports": [ ],
                     "resources": {
                        "limits": {
                           "cpu": "10m",
                           "memory": "16Mi"
                        },
                        "requests": {
                           "cpu": "10m",
                           "memory": "16Mi"
                        }
                     },
                     "volumeMounts": [ ]
                  }
               ],
               "volumes": [ ]
            }
         }
      }
   }
}
